src/Crawler.php

```diff
@@ -238,6 +260,8 @@ public function startCrawling($baseUrl)
 
         $crawlUrl = CrawlUrl::create($this->baseUrl);
 
+        $this->robotsTxt = $this->createRobotsTxt($crawlUrl->url);
```

Comment:
Will this work when the given url is not the root of a domain?
eg `https://example.com/page/subpage`

Reply:
Yes, the URL is used in `createRobotsTxt` like so:

```
$uri->withPath('/robots.txt')
```

By the robots spec, the `robots.txt` must always be in the root.